plugins/promela/hu.bme.mit.gamma.xsts.promela.transformation/src/hu/bme/mit/gamma/xsts/promela/transformation/serializer/ModelSerializer.xtend

```diff
@@ -146,4 +177,97 @@ class ModelSerializer {
 			«subaction.serialize»
 		«ENDFOR»
 	'''
+	
+	def dispatch String serialize(ParallelAction action) {
+		val actionSize = action.actions.size
+		val actions = action.actions
```

Comment:
Extract action.actions into a separate local variable (to avoid code duplication).